### PR TITLE
ENH: implement flutter detection

### PR DESCRIPTION
**Pull request by nrwslac**
## Description

- Implement a function `FB_FlutterDetection` to detect a boolean changing from high to low a given amount of times within a certain time window.
- Can detect a boolen flipping state number of cycles - 1 times.
- Add two tests:
- `TestFlutterReset`
- `TestFlutterResolution`
- Reorg tests into one central location.

## Motivation and Context

- https://jira.slac.stanford.edu/browse/ECS-3967
- MR1K4 was intermittently declaring a good state then a bad state blocking beam. The fluttering made it hard to bypass. 
## How Has This Been Tested?

- Two tests run and passed. Although, the actions I expected to happen on a particular cycle did not line up exactly. This means there's probably + 30 ms of  uncertainty in how long it takes to reset the flutter detection and maybe +10 ms in flutter detection. I don't anticipate this level of performance being needed atm.

## Pre-merge checklist
- [ ] Code works interactively
- [ ] Code contains descriptive comments
- [ ] Test suite passes locally
- [ ] Libraries are set to ``Always Newest`` version (``Library, *``)
- [ ] Committed with ``pre-commit`` or ran ``pre-commit run --all-files``

### CODE REVIEW

File: LCLSGeneral/LCLSGeneral/LCLSGeneral.plcproj [hunk 1]
```diff
@@ -17,7 +17,7 @@
     <CombineIds>false</CombineIds>
     <Company>SLAC</Company>
     <Released>false</Released>
-    <Title>LCLS General</Title>
+    <Title>LCLS General-nrw</Title>
```
Comment (ZLLentz) [inline LCLSGeneral/LCLSGeneral/LCLSGeneral.plcproj:20]: Need to switch back before merge

**Review by ZLLentz — COMMENTED** [review]
I think this looks solid :+1:
I followed the logic for the FB and for the tests
I had one comment about the dev library rename

File: LCLSGeneral/LCLSGeneral/POUs/Tests/FB_FlutterDetection_Test.TcPOU [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+]]></Declaration>
+      <Implementation>
+        <ST><![CDATA[(*Test the max flips that can be detected, should be number of cycles - 1*)
+
+TEST('TestFlutterResolution');
+fbTestTimer(IN := TRUE);
+
+nCycleTime := _TaskInfo[GETCURTASKINDEXEX()].CycleTime;
+nCycleTime := nCycleTime / 10000; //convert to ms
+nCalcCycleTime := UDINT_TO_TIME(nCycleTime * 4);
```
Comment (nrwslac) [inline LCLSGeneral/LCLSGeneral/POUs/Tests/FB_FlutterDetection_Test.TcPOU:92]: I think I missed setting this magic 4 to `nNumCycles`

Comment (ZLLentz) [reply LCLSGeneral/LCLSGeneral/POUs/Tests/FB_FlutterDetection_Test.TcPOU:92]: If the tests are really precise in what they're expecting we might get different behavior if we switch target PLCs

Comment (nrwslac) [reply LCLSGeneral/LCLSGeneral/POUs/Tests/FB_FlutterDetection_Test.TcPOU]: replaced 4 with variable.

File: LCLSGeneral/LCLSGeneral/LCLSGeneral.plcproj (hunk 1 above)
Comment (nrwslac) [reply LCLSGeneral/LCLSGeneral/LCLSGeneral.plcproj]: switched back to original project name.

**Review by ZLLentz — APPROVED** [review]
I'm fully :+1 on this, good job!

**Pull request merged**